download/download.go

```diff
@@ -405,6 +405,13 @@ func (d *Downloader) download(ctx context.Context, m metrics.Metrics) (*download
 			longPoll: d.longPollingEnabled,
 		}, nil
 	default:
+		if d.logger.GetLevel() == logging.Debug && resp.Body != nil {
```

Comment:
The docs for `http.Response.Body` states: 

```
The http Client and Transport guarantee that Body is always
non-nil, even on responses without a body or responses with
a zero-length body.
```

But I don't see any harm in being cautious and do a `nil` check anyways.